pkg/api/metadata/payer_info.go

```diff
@@ -0,0 +1,87 @@
+package metadata
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/xmtp/xmtpd/pkg/db/queries"
+	"github.com/xmtp/xmtpd/pkg/proto/xmtpv4/metadata_api"
+	"github.com/xmtp/xmtpd/pkg/utils"
+)
+
+type PayerInfoGroupBy string
+
+const (
+	PayerInfoGroupByHour PayerInfoGroupBy = "hour"
+	PayerInfoGroupByDay  PayerInfoGroupBy = "day"
+)
+
+type IPayerInfoFetcher interface {
+	GetPayerByAddress(ctx context.Context, address string) (int32, error)
+	GetPayerInfo(
+		ctx context.Context,
+		payerID int32,
+		afterTimestamp,
+		beforeTimestamp time.Time,
+		groupBy PayerInfoGroupBy,
+	) (*metadata_api.GetPayerInfoResponse_PayerInfo, error)
+}
+
+type PayerInfoFetcher struct {
+	queries *queries.Queries
+}
+
+func NewPayerInfoFetcher(db *sql.DB) *PayerInfoFetcher {
+	return &PayerInfoFetcher{
+		queries: queries.New(db),
+	}
+}
+
+// Gets the total spend and message count for a payer between the two timestamps, grouped by the appropriate granularity.
+func (f *PayerInfoFetcher) GetPayerInfo(
+	ctx context.Context,
+	payerID int32,
+	afterTimestamp,
```

Comment:
I always get confused by the fact that Go allows omitting the type if multiple types are the same. we generally specify the type for every parameter